#238 When calculating payouts, take all previous months into account.

Alfredo-Delgado opened the pull request:
Resolves #238

justinbmeyer commented:
Thanks for working on this, but this isn't what I had in mind. The previous contribution months should be loaded to calculate the pot. 

Sent from my iPhone

> On Jan 9, 2017, at 2:31 PM, Alfredo Delgado <notifications@github.com> wrote:
> 
> …tribution month. Resolves #238
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/donejs/bitcentive/pull/240
> 
> Commit Summary
> 
> Carry over the previous month's contributions when creating a new contribution month. Resolves #238
> File Changes
> 
> M public/components/select-contribution-month/select-contribution-month.js (3)
> Patch Links:
> 
> https://github.com/donejs/bitcentive/pull/240.patch
> https://github.com/donejs/bitcentive/pull/240.diff
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

Alfredo-Delgado commented:
Ok, I believe this is work that has not been addressed, yet. It'd be related to #161

justinbmeyer commented:
I think the best short-term fix is to simply load every contribution month.  We probably need to also load the contributors to check ones that are active.

Alfredo-Delgado commented:
The calculations already took previous months into account, although they don't exclude contributors whose `isActive` bit is false. This PR gathers all unique contributors and projects up to the current month before calculating the payouts.

justinbmeyer commented on public/components/payouts/payouts.js:
```diff
@@ -6,88 +6,92 @@ import moment from "moment";
-    value() {
-      return ContributionMonth.getList({});
-    }
+  contributionMonth: {
```
`contributionMonth: ContributionMonth`

justinbmeyer commented on public/models/contribution-month/contribution-month.js:
```diff
@@ -266,6 +264,109 @@ ContributionMonth.List = DefineList.extend({
+
+    this.filter(contributionMonth => {
+      return currentMoment.isSameOrAfter(contributionMonth.date);
+    }).map(contributionMonth => {
```
`forEach`

justinbmeyer commented on public/models/contribution-month/contribution-month.js:
```diff
@@ -266,6 +264,109 @@ ContributionMonth.List = DefineList.extend({
+	 * @return {Map}
+   */
+  getMonthlyPayouts(currentMonth) {
+		let currentMoment = moment(currentMonth.date);
```
space issue?

Alfredo-Delgado commented:
- replaced `map` with `forEach`
- replaced all spaces with tabs in `contribution-month.js`
- renamed cloned contributor to `contributorPayout` for better readability
